src/ast/passes/semantic_analyser.cpp

```diff
@@ -3703,6 +3738,11 @@ bool SemanticAnalyser::check_call(const Call &call)
     return true;
   }
 
+  if (is_final_pass() && call.ret_val_discarded &&
+      spec->second.discard_ret_warn) {
+    call.addWarning() << "Return value discarded";
```

Comment:
Maybe a bit of exposition? I.e. explain a bit more about what the problem is:

`call.ident << " return value is discarded, it should be used";`